arbos/l2pricing/model_constraints.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package l2pricing
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/arbitrum/multigas"
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+func (ps *L2PricingStateConstraints) AddToResourceGasPool(resource multigas.ResourceKind, gas int64) error {
+	backlog, err := ps.GetResourceBacklog(resource)
+	if err != nil {
+		return err
+	}
+
+	if gas > 0 {
+		backlog = arbmath.SaturatingUSub(backlog, uint64(gas))
+	} else {
+		backlog = arbmath.SaturatingUAdd(backlog, uint64(-gas))
+	}
+
+	return ps.resourceBacklogs[resource].Set(backlog)
+}
+
+func (ps *L2PricingStateConstraints) UpdatePricingModel(l2BaseFee *big.Int, timePassed uint64, debug bool) {
+	ps.L2PricingState.UpdatePricingModel(l2BaseFee, timePassed, debug)
+
+	baseFee, _ := ps.BaseFeeWei()
+	minBaseFee, _ := ps.MinBaseFeeWei()
+	inertia, _ := ps.PricingInertia()
+
+	for resource, periodMap := range ps.constraints {
```

Comment:
It is a good idea to have a method to iterate over the resource constraints. Here we are exposing some internal details of the resource constraints type.